src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -209,4 +215,98 @@ public void parseTags_collectionWithValidTags_returnsTagSet() throws Exception {
 
         assertEquals(expectedTagSet, actualTagSet);
     }
+
+    @Test
+    public void parseAttribute_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseAttribute(null));
+    }
+
+    @Test
+    public void parseAttribute_unknownAttributeWithoutWhitespace_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseAttribute("gamer"));
+    }
+
+    @Test
+    public void parseAttribute_unknownAttributeWithWhitespace_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseAttribute(WHITESPACE + "gamer" + WHITESPACE));
+    }
+
```

Comment:
LGTM! might want to include testcases such that input `WHITESPACE + VALID_ATTRIBUTE_1 + WHITESPACE + "gamer"` would not be valid

Reply:
Additional testcases have been added!